healthcare/api-client/v1/fhir/fhir_resources_test.py

```diff
@@ -0,0 +1,304 @@
+            # The API returns 403 when the dataset doesn't exist.
+            if err.resp.status == 404 or err.resp.status == 403:
+                print("Got exception {} while deleting dataset".format(err.resp.status))
+            else:
+                raise
+
+    clean_up()
+
+
+@pytest.fixture(scope="module")
+def test_fhir_store():
+    @retry(
+        wait_exponential_multiplier=1000,
+        wait_exponential_max=10000,
+        stop_max_attempt_number=10,
+        retry_on_exception=retry_if_server_exception,
+    )
+    def create():
+        try:
+            fhir_stores.create_fhir_store(
+                project_id, cloud_region, dataset_id, fhir_store_id
+            )
+        except HttpError as err:
+            # We ignore 409 conflict here, because we know it's most
+            # likely the first request failed on the client side, but
+            # the creation suceeded on the server side.
+            if err.resp.status == 409:
+                print(
+                    "Got exception {} while creating FHIR store".format(err.resp.status)
+                )
+            else:
+                raise
+
+    create()
+
+    yield
+
+    # Clean up
+    @retry(
+        wait_exponential_multiplier=1000,
+        wait_exponential_max=10000,
+        stop_max_attempt_number=10,
+        retry_on_exception=retry_if_server_exception,
+    )
+    def clean_up():
+        try:
+            fhir_stores.delete_fhir_store(
+                project_id, cloud_region, dataset_id, fhir_store_id
+            )
+        except HttpError as err:
+            # The API returns 403 when the FHIR store doesn't exist.
+            if err.resp.status == 404 or err.resp.status == 403:
+                print(
+                    "Got exception {} while deleting FHIR store".format(err.resp.status)
+                )
+            else:
+                raise
+
+    clean_up()
+
+
+# Fixture that creates/deletes a Patient resource for various tests.
+@pytest.fixture(scope="module")
+def test_patient():
+    patient_response = fhir_resources.create_patient(
+        project_id, cloud_region, dataset_id, fhir_store_id,
+    )
+    patient_resource_id = patient_response["id"]
+
+    yield patient_resource_id
+
+    @retry(
+        wait_exponential_multiplier=1000,
+        wait_exponential_max=10000,
+        stop_max_attempt_number=10,
+        retry_on_exception=retry_if_server_exception,
+    )
+    # Clean up
+    def clean_up():
+        try:
+            fhir_resources.delete_resource(
+                project_id,
+                cloud_region,
+                dataset_id,
+                fhir_store_id,
+                resource_type,
+                patient_resource_id,
+            )
+
+        except HttpError as err:
+            # The API returns 200 whether the resource exists or was
+            # successfully deleted or not, so only retry on
+            # unathorized exceptions.
+            if err.resp.status == 401:
```

Comment:
This code ignores the exception when the status is 401. Is this intended?

Reply:
Code removed.

Reply:
Sorry I still see it